packages/x-charts/src/ChartsGrid/ChartsGrid.tsx

```diff
@@ -0,0 +1,127 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import composeClasses from '@mui/utils/composeClasses';
+import { styled } from '@mui/material/styles';
+
+import { CartesianContext } from '../context/CartesianContextProvider';
+import useTicks from '../hooks/useTicks';
+import { ChartsGridClasses, getGridUtilityClass, gridClasses } from './gridClasses';
+
+export const GridRoot = styled('g', {
+  name: 'MuiChartsGrid',
+  slot: 'Root',
+  overridesResolver: (props, styles) => styles.root,
+})(({ theme }) => ({
+  [`& .${gridClasses.line}`]: {
+    stroke: (theme.vars || theme).palette.divider,
+    shapeRendering: 'crispEdges',
+    strokeWidth: 1,
+  },
+}));
+
+const useUtilityClasses = ({ classes }: ChartsGridProps) => {
+  const slots = {
+    root: ['root'],
+    verticalLine: ['line', 'verticalLine'],
+    horizontalLine: ['line', 'horizontalLine'],
+  };
+
+  return composeClasses(slots, getGridUtilityClass, classes);
+};
+
+export interface ChartsGridProps {
+  /**
+   * Displays vertical grid.
+   */
+  vertical?: boolean;
+  /**
+   * Displays horizontal grid.
+   */
+  horizontal?: boolean;
+  /**
+   * Override or extend the styles applied to the component.
+   */
+  classes?: Partial<ChartsGridClasses>;
+}
+
+/**
+ * Demos:
+ *
+ * - [Axis](https://mui.com/x/react-charts/axis/)
+ *
+ * API:
+ *
+ * - [ChartsGrid API](https://mui.com/x/api/charts/charts-axis/)
+ */
+function ChartsGrid(props: ChartsGridProps) {
+  const { vertical, horizontal, ...other } = props;
+  const { xAxis, xAxisIds, yAxis, yAxisIds } = React.useContext(CartesianContext);
+
+  const classes = useUtilityClasses(props);
+
+  const horizontalAxisId = yAxisIds[0];
+  const verticalAxisId = xAxisIds[0];
+
+  const {
+    scale: xScale,
+    tickNumber: xTickNumber,
+    tickInterval: xTickInterval,
+  } = xAxis[verticalAxisId];
+
+  const {
+    scale: yScale,
+    tickNumber: yTickNumber,
+    tickInterval: yTickInterval,
+  } = yAxis[horizontalAxisId];
+
+  const xTicks = useTicks({ scale: xScale, tickNumber: xTickNumber, tickInterval: xTickInterval });
+  const yTicks = useTicks({ scale: yScale, tickNumber: yTickNumber, tickInterval: yTickInterval });
```

Comment:
I'm relying on the fact that the grid use the same parametters and the same hook as axies ticks to get the alignement.

I though about doing the computation once and let axes and grid rely on it. But the benefit is not clear whereas the effort is important. So I decided to go that way, assuming it will be enough for at least a year, and if needed we will be able to review this process in the next major, the time to accumulate feedback